rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java

```diff
@@ -816,8 +817,17 @@ public HttpSenderResponseException(@Nullable Throwable cause, @Nullable Integer
             this.responseCode = responseCode;
         }
 
+        /**
+         * All 400s are considered client-side exceptions, but we only want to cache ones that are unlikely to change
+         * if requested again in order to save on time spent making HTTP calls.
+         * <br/>
+         * For 408 TIMEOUT, 425 TOO_EARLY, and 429 TOO_MANY_REQUESTS, these are likely to change if not cached.
+         */
         public boolean isClientSideException() {
-            return responseCode != null && responseCode >= 400 && responseCode <= 404;
+            if (responseCode == null || responseCode < 400 || responseCode > 499) {
```

Comment:
Again; a preference to show visually that responseCode is outside a range.
```suggestion
            if (responseCode == null || responseCode < 400 || 499 < responseCode) {
```